docs/source/1.0/guides/migrationg-idl-1-to-2.rst

```diff
@@ -0,0 +1,292 @@
+        output: PutUserOutput
+    }
+
+    @input
+    structure PutUserInput {
+        email: String,
+        id: String,
+        username: String,
+        description: String
+    }
+
+    @output
+    structure PutUserOutput {}
+
+can be updated to:
+
+.. code-block::
+
+    $version: "2.0"
+
+    namespace smithy.example
+
+    operation PutUser {
+        input := {
+            email: String
+            id: String
+            username: String
+            description: String
+        },
+        output := {}
+    }
+
+.. seealso::
+
+    the :ref:`inline input / output <idl-inline-input-output>` section of the
+    spec for more details.
+
+
+Abstract shared shape configuration with mixins
+-----------------------------------------------
+
+Models often have several shapes that refer to the same sets of members, or
+which share a set of trait configurations. For example, resource instance
+operations all require that the resource's identifiers be present. With
+:ref:`mixins`, it is easy to simply share these member definitions without
+having to copy and paste them. The following model:
+
+.. code-block:: smithy
+
+    $version: "1.0"
+
+    namespace smithy.example
+
+    resource User {
+        identifiers: {
+            email: String,
+            id: String,
+            username: String
+        },
+        read: GetUser
+    }
+
+    operation GetUser {
+        input: GetUserInput,
+        output: GetUserOutput
+    }
+
+    @input
+    structure GetUserInput {
+        email: String,
+        id: String,
+        username: String
+    }
+
+    @output
+    structure GetUserOutput {
+        email: String,
+        id: String,
+        username: String,
+        description: String
+    }
+
+Can be updated to:
+
+.. code-block:: smithy
+
+    $version: "2.0"
+
+    namespace smithy.example
+
+    resource User {
+        identifiers: {
+            email: String
+            id: String
+            username: String
+        },
+        read: GetUser
+    }
+
+    @mixin
+    structure UserIdentifiers {
+        email: String
+        id: String
+        username: String
+    }
+
+    operation GetUser {
+        input := with [UserIdentifiers] {}
+        output := with [UserIdentifiers] {
+            description: String
+        }
+    }
+
+Similarly, :ref:`mixins` can be useful if you have a shared set of traits
+that otherwise have to be copied and pasted.
+
+.. seealso::
+
+    the :ref:`mixins section <mixins>` of the spec for more details on how they
+    work.
+
+
+Remove unsightly commas
+-----------------------
+
+Smithy IDL 2.0 removed the need to include commas when defining, lists, maps,
+and shape properties. For example, the following model:
+
+.. code-block:: smithy
+
+    $version: "1.0"
+
+    namespace smithy.example
+
+    operation GetUser {
+        input: GetUserInput,
+        output: GetUserOutput,
+        errors: [
+            NotFoundError,
+            AccessDeniedError,
+        ],
+    }
+
```

Comment:
Nit: can just remove this line and the next in both this example and the next example